### PR TITLE
Add more java tests

### LINKED ISSUE #26: Add more tests for the Java API (Functional)

### PR DESCRIPTION
**Pull request by VottonDev**
Closes #26

### CODE REVIEW

**Comment by VottonDev** [comment]
When writing tests, note that it will be harder to write them for things such as interruptions as requires imports such as interruption groups, deviceDAO as interruptions themselves from the looks of it require a group to belong to and a device.

Next, the easiest test would be probably to create a new Keycloak user, deleting it and similar as won't require many imports.

File: jouko-api/src/main/java/fi/metatavu/jouko/api/dao/ControllerDAO.java [hunk 1]
```diff
@@ -41,8 +36,9 @@ public ControllerEntity create(String eui, String key, ControllerCommunicationCh
    * Deletes the controller entity with the given ID.
    *
    * @param id the ID of the controller entity to delete.
+   * @return
    */
-  public void delete(Long id) {
+  public ControllerEntity delete(Long id) {
```
Comment (VottonDev) [inline jouko-api/src/main/java/fi/metatavu/jouko/api/dao/ControllerDAO.java:41]: I think changed this so it works with the tests but have to be careful with this, so it doesn't break anything else.

Comment (VottonDev) [reply jouko-api/src/main/java/fi/metatavu/jouko/api/dao/ControllerDAO.java:41]: Nvm, it's okay. Read the rest of the code and it's meant to have the ControllerEntity in it but for some reason originally it didn't but it wasn't utilised at all nonetheless.

File: jouko-api/src/main/java/fi/metatavu/jouko/api/dao/ControllerDAO.java [hunk 2]
```diff
@@ -56,6 +52,7 @@ public void delete(Long id) {
     );
 
     em.createQuery(delete).executeUpdate();
+    return null;
```
Comment (VottonDev) [inline jouko-api/src/main/java/fi/metatavu/jouko/api/dao/ControllerDAO.java:55]: The code wasn't returning anything, so the functions utilising it such as tests wouldn't know if it successfully worked or not.

File: jouko-api/src/test/java/fi/metatavu/jouko/api/dao/InterruptionGroupDAOTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    //      */
+    //     Mockito.when(interruptionGroupDAO.findById(1L)).thenReturn(group);
+    //     Assert.assertEquals(group, interruptionGroupDAO.findById(1L));
+
+    //     /**
+    //      * Update interruption group
+    //      */
+    //     OffsetDateTime startTimeUpdated = OffsetDateTime.ofInstant(Instant.ofEpochSecond(200), ZoneOffset.UTC);
+    //     OffsetDateTime endTimeUpdated = OffsetDateTime.ofInstant(Instant.ofEpochSecond(300), ZoneOffset.UTC);
+    //     interruptionGroupDAO.update(group, startTimeUpdated, endTimeUpdated);
```
Comment (VottonDev) [inline jouko-api/src/test/java/fi/metatavu/jouko/api/dao/InterruptionGroupDAOTest.java]: Doesn't update the startTime and endTime for some reason, it's as if the update method itself doesn't work at all as logging it returns nothing but doesn't produce any errors either.

File: jouko-api/src/test/java/fi/metatavu/jouko/api/dao/InterruptionDAOTest.java [hunk 4]
```diff
@@ -64,4 +65,18 @@ public void deleteInterruption() {
...
+     * List interruptions by device and date (DeviceEntity device, OffsetDateTime start, OffsetDateTime end)
+     */
+    @Test
+    public void listInterruptionsByDeviceAndDate() {
+        UserEntity user = new UserEntity(1L,"keycloakId", "name");
+        ControllerEntity controller = new ControllerEntity(1L, "EUI", "KEY", ControllerCommunicationChannel.GPRS);
+        DeviceEntity device = new DeviceEntity(1L, "Device", user, controller);
+        InterruptionGroupEntity group = new InterruptionGroupEntity(1L, OffsetDateTime.ofInstant(Instant.ofEpochSecond(0), ZoneOffset.UTC), OffsetDateTime.ofInstant(Instant.ofEpochSecond(100), ZoneOffset.UTC));
+        InterruptionEntity interruption = interruptionDAO.create(device, group);
+        Mockito.when(interruptionDAO.listByDeviceAndDate(device, OffsetDateTime.ofInstant(Instant.ofEpochSecond(0), ZoneOffset.UTC), OffsetDateTime.ofInstant(Instant.ofEpochSecond(100), ZoneOffset.UTC))).thenReturn(Arrays.asList());
```
Comment (VottonDev) [inline jouko-api/src/test/java/fi/metatavu/jouko/api/dao/InterruptionDAOTest.java]: This might be the wrong method but will look further into today.

**Comment by VottonDev** [comment]
I think all the tests have now been roughly written, so pretty much cover all the DAOs.

The only thing that doesn't work is the User deletion test for some reason, the UserEntity doesn't disappear after running the UserDAO.delete but it could be an issue with the method alone but it doesn't output any errors or anything at all.

Edit: Maybe needs a separate method to deal with it as the UserEntity is a bit more complex because of Keycloak, so #46 is relevant to that.

**Review by ghost — APPROVED** [review]

**Pull request merged**